### PR TITLE
[16.0][MIG] rma_sale: Migration to 16.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
- Standard procedure.
- Adapt to new `prepare_*_vals` methods.
- Renamed fields adaptation.

@Tecnativa TT44214

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Backend functionality ok. Some issues I came into:

The wizard form window doesn't show up properly

image

Looks like the portal wizard backend js code needs to be retailored as it doesn't work properly. Do you want me to take a look into it?

**Comment by pedrobaeza** [comment]
Ouch, I looked for o2m in views, but not in wizards. Fixed.

About the portal JS, please do the needed changes, and it would be good to have a tour test for checking it.

File: rma_sale/views/sale_portal_template.xml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                </div>
+                <t
+                    t-set="delivery_addresses"
+                    t-value="sale_order.partner_shipping_id | sale_order.partner_id.commercial_partner_id.mapped('child_ids').filtered(lambda x: x.type in ['contact', 'delivery'])"
+                />
+                <button
+                    class="btn btn-primary btn-block mb8"
+                    type="button"
+                    data-toggle="collapse"
+                    data-target="#delivery_address_picker"
```
Comment (chienandalu) [inline rma_sale/views/sale_portal_template.xml]:
```suggestion
                    data-bs-toggle="collapse"
                    data-bs-target="#delivery_address_picker"
```

File: rma_sale/views/sale_portal_template.xml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                    aria-expanded="false"
+                ><i class="fa fa-truck" /> Choose a delivery address</button>
+                <div class="col-lg-12 collapse mt8" id="delivery_address_picker">
+                    <div data-toggle="buttons" class="row">
```
Comment (chienandalu) [inline rma_sale/views/sale_portal_template.xml]:
```suggestion
                    <div data-bs-toggle="buttons" class="row">
```

File: rma_sale/views/sale_portal_template.xml [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                                                    <t t-out="operation.name" />
+                                                </option>
+                                            </t>
+                                        </select>
+                                    </td>
+                                    <td>
+                                        <button
+                                            class="btn btn-primary fa fa-comments"
+                                            type="button"
+                                            data-toggle="collapse"
```
Comment (chienandalu) [inline rma_sale/views/sale_portal_template.xml]:
```suggestion
                                            data-bs-toggle="collapse"
```

File: rma_sale/views/sale_portal_template.xml [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+                    >
+                    <i class="fa fa-reply" /> Request RMAs
+                </a>
+            </t>
+            <t t-else="">
+                <a
+                        role="button"
+                        class="btn btn-secondary btn-block mb8"
+                        data-toggle="modal"
+                        data-target="#modal-request-rma"
```
Comment (chienandalu) [inline rma_sale/views/sale_portal_template.xml]:
```suggestion
                        data-bs-toggle="modal"
                        data-bs-target="#modal-request-rma"
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Finally no js revamp was needed but just some bs5 attributes tweaking

**Comment by chienandalu** [comment]
I just tried this PR https://github.com/OCA/openupgradelib/pull/338 to migrate the qweb templates and it looks that its working nicely :) I added a commit to the PR

File: rma_sale/views/sale_portal_template.xml [hunk 5]
```diff
@@ -61,15 +61,15 @@
                                 type="radio"
                                 name="partner_shipping_id"
                                 t-att-value="address.id"
-                            >
+                            />
```
Comment (chienandalu) [inline rma_sale/views/sale_portal_template.xml]: <strike>This change is weird. I guess, the bs4>bs5 scripts need some tweaking :) I'll fix it by hand</strike>

<strike>Ah, no, indeed it's correct to close it...</strike>

I'll check functionally

Comment (chienandalu) [reply rma_sale/views/sale_portal_template.xml]: Ok, now I remember why this was strange. This was a bs4 hack to select the rma address from the address cards with no js. I'll check how to adapt it to make it work, but the change was correct, I think.

**Comment by pedrobaeza** [comment]
Is it correct then? Are you going to prepare a tour test?

**Comment by chienandalu** [comment]
> Is it correct then? Are you going to prepare a tour test?

No, I should fix the delivery address selector in the portal. I can prepare a tour, yes.

**Review by chienandalu — APPROVED** [review]
- portal functionality fixed
- tours added

**Comment by pedrobaeza** [comment]
Is this finished then?

**Comment by chienandalu** [comment]
Yes :)

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I have improved a bit more the tests coverage and fine-tune several warnings, so it's time to merge it:

/ocabot merge nobump

**Pull request merged**